code/modules/antagonists/abductor/equipment/glands/heal.dm

```diff
@@ -0,0 +1,177 @@
+		playsound(owner, 'sound/effects/splat.ogg', 50, 1)
+		eyes.Remove(owner)
+		eyes.forceMove(owner.drop_location())
+	else
+		to_chat(owner, "<span class='warning'>You feel a weird rumble behind your eye sockets...</span>")
+
+	addtimer(CALLBACK(src, .proc/finish_replace_eyes), rand(100, 200))
+
+/obj/item/organ/heart/gland/heal/proc/finish_replace_eyes()
+	var/eye_type = /obj/item/organ/eyes
+	if(owner.dna.species && owner.dna.species.mutanteyes)
+		eye_type = owner.dna.species.mutanteyes
+	var/obj/item/organ/eyes/new_eyes = new eye_type()
+	new_eyes.Insert(owner)
+	owner.visible_message("<span class='warning'>A pair of new eyes suddenly inflates into [owner]'s eye sockets!</span>", "<span class='userdanger'>A pair of new eyes suddenly inflates into your eye sockets!</span>")
+
+/obj/item/organ/heart/gland/heal/proc/replace_limb(body_zone, obj/item/bodypart/limb)
+	if(limb)
+		owner.visible_message("<span class='warning'>[owner]'s [limb.name] suddenly detaches from [owner.p_their()] body!</span>", "<span class='userdanger'>Your [limb.name] suddenly detaches from your body!</span>")
+		playsound(owner,pick('sound/misc/desceration-01.ogg','sound/misc/desceration-02.ogg','sound/misc/desceration-01.ogg') ,50, 1, -1)
+		limb.drop_limb()
+	else
+		to_chat(owner, "<span class='warning'>You feel a weird tingle in your [parse_zone(body_zone)]... even if you don't have one.</span>")
+
+	addtimer(CALLBACK(src, .proc/finish_replace_limb, body_zone), rand(150, 300))
+	
+/obj/item/organ/heart/gland/heal/proc/finish_replace_limb(body_zone)
+	owner.visible_message("<span class='warning'>With a loud snap, [owner]'s [parse_zone(body_zone)] rapidly grows back from [owner.p_their()] body!</span>", 
+	"<span class='userdanger'>With a loud snap, your [parse_zone(body_zone)] rapidly grows back from your body!</span>",
+	"<span class='warning'>Your hear a loud snap.</span>")
+	playsound(owner, 'sound/magic/demon_consume.ogg', 50, 1)
+	owner.regenerate_limb(body_zone)
+
+/obj/item/organ/heart/gland/heal/proc/replace_blood()
+	owner.visible_message("<span class='warning'>[owner] starts vomiting huge amounts of blood!</span>", "<span class='userdanger'>You suddenly start vomiting huge amounts of blood!</span>")
+	keep_replacing_blood()
+
+/obj/item/organ/heart/gland/heal/proc/keep_replacing_blood()
+	var/keep_going = FALSE
+	owner.vomit(0, TRUE, FALSE, 3, FALSE, FALSE, FALSE, TRUE)
+	owner.Stun(15)
+	owner.adjustToxLoss(-15, TRUE, TRUE)
+
+	owner.blood_volume = min(BLOOD_VOLUME_NORMAL, owner.blood_volume + 20)
+	if(owner.blood_volume < BLOOD_VOLUME_NORMAL)
+		keep_going = TRUE
+
+	if(owner.getToxLoss())
+		keep_going = TRUE
+	for(var/datum/reagent/toxin/R in owner.reagents.reagent_list)
+		owner.reagents.remove_reagent(R.type, 4)
+		if(owner.reagents.has_reagent(R.type))
+			keep_going = TRUE
+	if(keep_going)
+		addtimer(CALLBACK(src, .proc/keep_replacing_blood), 30)
```

Comment:
Doesn't look like this gets stopped by death. It also might keep racking up timer stacks every time activate gets called and the conditions are not met.